main.go

```diff
@@ -0,0 +1,257 @@
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-tools/go-steputils/input"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+)
+
+// config ...
+type config struct {
+	SourcePath string `env:"source_path,required"`
+	TargetDir  string `env:"target_dir"`
+}
+
+func main() {
+	var cfg config
+	if err := stepconf.Parse(&cfg); err != nil {
+		log.Errorf("Error: %s\n", err)
+		os.Exit(1)
+	}
+	stepconf.Print(cfg)
+
+	fmt.Println()
+	cfg.print()
+
+	if err := cfg.validate(); err != nil {
+		fail("Issue with input: %s", err)
+	}
+
+	_, err := ensureZIPExtension(cfg.SourcePath, cfg.TargetDir)
+
+	if err != nil {
+		fail("Issue with compress: %s", err)
+		return
+	}
+}
+
+// fixTargetExtension Adds the file extension to the end of the targetpath if it's needed.
+func fixTargetExtension(targetDir string) string {
+	targetExtension := ".zip"
+
+	if strings.HasSuffix(targetDir, targetExtension) {
+		return targetDir
+	}
+	return fmt.Sprintf("%s%s", targetDir, targetExtension)
+}
+
+func ensureZIPExtension(sourcePath string, targetDir string) (string, error) {
+	targetDir = fixTargetExtension(targetDir)
+
+	ensureDir(targetDir)
+
+	zipfile, err := os.Create(targetDir)
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		if err = zipfile.Close(); err != nil {
+			log.Errorf("%s", err)
+		}
+	}()
+
+	archive := zip.NewWriter(zipfile)
+
+	defer func() {
+		if err = archive.Close(); err != nil {
+			log.Errorf("%s", err)
+		}
+	}()
+
+	info, err := os.Lstat(sourcePath)
+	if err != nil {
+		return "", err
+	}
+
+	var baseDir string
+	if !info.IsDir() {
+		baseDir = filepath.Base(sourcePath)
+	}
+
+	err = filepath.Walk(sourcePath, func(pth string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		isSymlink, evaledPath, err := checkSymlink(pth)
+		if err != nil {
+			return err
+		}
+
+		originalPath := pth
+
+		if isSymlink {
+			pth = evaledPath
+		}
+
+		info, err = os.Lstat(pth)
+		if err != nil {
+			return err
+		}
+
+		header, err := zip.FileInfoHeader(info)
+		if err != nil {
+			return err
+		}
+
+		if baseDir != "" {
+			if isSymlink {
+				header.Name = filepath.Join(baseDir, strings.TrimPrefix(originalPath, sourcePath))
+			} else {
+				header.Name = filepath.Join(baseDir, strings.TrimPrefix(pth, sourcePath))
+			}
+
+		} else {
+			baseDir = "/"
+		}
+
+		if info.IsDir() {
+			header.Name += "/"
+		} else {
+			header.Method = zip.Deflate
+		}
+
+		writer, err := archive.CreateHeader(header)
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		file, err := os.Open(pth)
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			if err = file.Close(); err != nil {
+				log.Errorf("%s", err)
+			}
+		}()
+
+		_, err = io.Copy(writer, file)
+		return err
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprint(zipfile), nil
+}
+
+// If the file is a symbolic link it will evaulate the path name.
```

Comment:
Start the comment with the name of the function, and don't use newlines for formatting reasons, the godoc will wrap the text properly.